api-tests/src/test/java/org/openmrs/module/reporting/cohort/definition/evaluator/DrugOrderCohortDefinitionEvaluatorTest.java

```diff
@@ -0,0 +1,185 @@
+  }
+
+  @After
+  public void tearDown() {
+    cohortDefinition = null;
+  }
+  
+  @Test
+  public void evaluate_shouldReturnAllPatientsCurrentlyActiveOnDrugs() throws Exception {
+    cohortDefinition.setOnlyCurrentlyActive(true);
+    Cohort cohort = Context.getService(CohortDefinitionService.class).evaluate(cohortDefinition, null);
+    System.out.println("Currently on Drugs " + cohort.getMemberIds() + " " + cohortDefinition + "===");
+  }
+  
+  @Test
+  public void evaluate_shouldReturnAllPatientsCurrentlyNotActiveOnDrugs() throws Exception {
+    cohortDefinition.setOnlyCurrentlyActive(false);
+    Cohort cohort = Context.getService(CohortDefinitionService.class).evaluate(cohortDefinition, null);
+    System.out.println("Currently Not on Drugs " + cohort.getMemberIds() + " " + cohortDefinition + "===");
+  }
+  
+  @Test
+  public void evaluate_shouldReturnAllPatientsThatHaveTakenAnyofListedDrugs() throws Exception {
+    List drugSetList = new ArrayList<Concept>();
+    drugSetList.add(new Concept(88));
+    drugSetList.add(new Concept(792));
+    drugSetList.add(new Concept(20));
+    cohortDefinition.setDrugSets(drugSetList);
+    cohortDefinition.setWhich("ANY");
+    Cohort cohort = Context.getService(CohortDefinitionService.class).evaluate(cohortDefinition, null);
+    System.out.println("Taking Drugs Any" + cohort.getMemberIds()  + " " + cohortDefinition+ "===");
+    Assert.assertTrue(cohort.contains(2));
+    Assert.assertTrue(cohort.contains(21));
+    Assert.assertTrue(cohort.contains(22));
+    Assert.assertEquals(3, cohort.size());
+
+  }
+  
+  @Test
+  public void evaluate_shouldReturnAllPatientsThatHaveNeverTakenListedDrugs() throws Exception {
+    List drugSetList = new ArrayList<Concept>();
+    drugSetList.add(new Concept(88));
+    drugSetList.add(new Concept(792));
+    cohortDefinition.setDrugSets(drugSetList);
+    cohortDefinition.setWhich("NONE");
+    Cohort cohort = Context.getService(CohortDefinitionService.class).evaluate(cohortDefinition, null);
+    System.out.println("Taking Drugs None" + cohort.getMemberIds() +  " " + cohortDefinition + "===");
+    Assert.assertTrue(cohort.contains(21));
+    Assert.assertTrue(cohort.contains(22));
+    Assert.assertEquals(2, cohort.size());
+  }
+  
+  @Test
+  public void evaluate_shouldReturnAllPatientsThatHaveTakenAllListedDrugs() throws Exception {
+    List drugSetList = new ArrayList<Concept>();
+    drugSetList.add(new Concept(88));
+    drugSetList.add(new Concept(792));
+    cohortDefinition.setDrugSets(drugSetList);
+    cohortDefinition.setWhich("ALL");
+    Cohort cohort = Context.getService(CohortDefinitionService.class).evaluate(cohortDefinition, null);
+    System.out.println("Taking Drugs ALL" + cohort.getMemberIds() + " " + cohortDefinition);
+    Assert.assertTrue(cohort.contains(2));
+    Assert.assertEquals(1, cohort.size());
+
+  }
+
+  // @Test
+  // public void evaluate_shouldReturnAllPatientsTakingDrugsContainingIngredients() throws Exception {
```

Comment:
Did you take a look at this? https://wiki.openmrs.org/display/docs/Java+Conventions#JavaConventions-Donotcommentoutcode